plugins/atomicfu/atomicfu-compiler/src/org.jetbrains.kotlinx.atomicfu.compiler/extensions/AtomicFUTransformer.kt

```diff
@@ -0,0 +1,595 @@
+                            condition = condition.transformAtomicFunctionCall(parentDeclaration, lambda)
+                            result = result.transformAtomicFunctionCall(parentDeclaration, lambda)
+                        }
+                    }
+                }
+            }
+            is IrTry -> {
+                return apply {
+                    tryResult = tryResult.transformAtomicFunctionCall(parentDeclaration, lambda)
+                    catches.forEach {
+                        it.result = it.result.transformAtomicFunctionCall(parentDeclaration, lambda)
+                    }
+                    finallyExpression = finallyExpression?.transformAtomicFunctionCall(parentDeclaration, lambda)
+                }
+            }
+            is IrBlock -> {
+                return apply {
+                    statements.forEachIndexed { i, stmt ->
+                        statements[i] = stmt.transformStatement(parentDeclaration, lambda)
+                    }
+                }
+            }
+            is IrGetValue -> {
+                if (lambda != null && symbol.owner.parent == lambda) return this
+                if (symbol is IrValueParameterSymbol && parentDeclaration.isTransformedAtomicExtensionFunction()) {
+                    // replace use site of the value parameter with it's copy from the transformed declaration
+                    val index = (symbol.owner as IrValueParameter).index
+                    if (index >= 0) { // index == -1 for `this` parameter
+                        val transformedValueParameter = (parentDeclaration as IrFunction).valueParameters[index]
+                        return buildGetValue(transformedValueParameter.symbol)
+                    }
+                }
+            }
+            is IrCall -> {
+                dispatchReceiver?.let { dispatchReceiver = it.transformAtomicFunctionCall(parentDeclaration, lambda) }
+                extensionReceiver?.let { extensionReceiver = it.transformAtomicFunctionCall(parentDeclaration, lambda) }
+                getValueArguments().forEachIndexed { i, arg ->
+                    putValueArgument(i, arg?.transformAtomicFunctionCall(parentDeclaration, lambda))
+                }
+                val isInline = symbol.owner.isInline
+                val receiver = extensionReceiver ?: dispatchReceiver ?: return this
+                // Invocation of atomic functions on atomic receivers
+                // are substituted with the corresponding inline declarations from `kotlinx-atomicfu-runtime`
+                if (symbol.isKotlinxAtomicfuPackage() && receiver.type.isAtomicValueType()) {
+                    // 1. Receiver is the atomic field:
+                    // a.incrementAndGet() -> atomicfu_incrementAndGet(get_a {..}, set_a {..})
+                    if (receiver is IrCall) { // property accessor <get-field>
+                        val accessors = receiver.getAccessors(lambda ?: parentDeclaration)
```

Comment:
Why it has to be an accessor? I could always make an extension function with receiver of `AtomicValue` type